internal/heartbeat/heartbeat.go

```diff
@@ -37,7 +39,7 @@ type HeartbeatData struct {
 	hardware                        hw.Hardware
 	osInfo                          *os2.OS
 	previousMutableHardwareInfo     *models.HardwareInfo
-	previousMutableHardwareInfoLock sync.Mutex
+	previousMutableHardwareInfoLock sync.RWMutex
```

Comment:
WDYM? `previousMutableHardwareInfo` has to be protected to avoid data races when read/write accesses
Or do you mean I should use sync/atomic [LoadPointer](https://pkg.go.dev/sync/atomic#LoadPointer) and [StorePointer](https://pkg.go.dev/sync/atomic#StorePointer) function instead?

Reply:
exactly. load and store. but it does not really matter